summaries/search-based-survey.md

```diff
@@ -0,0 +1,335 @@
+# Search-Based Software Test Data Generation
+**Article:**
+*Phil McMinn*. Search-Based Software Test Data Generation: A Survey. Software Testing, Verification and Reliability, vol. 14, no. 2, pp. 105–156, 2004.
+
+## Metaheursitic Search Applications
+**Goals:**
+ - Coverage of the code.
+ - Exercise a specific feature of the program.
+ - Disprove grey-box properties.
+ - Verify non-functional properties.
+
+## Metaheursitic Search Techniques
+Solution should be encoded such that they can be manipulated by the search and 'neighbours' can be compared. 
+Using an *objective function*, the search distinguishes *good* solutions from *bad* solutions and guides the search using this function.
+
+### Hill Climbing
+This is a local search algorithm that starts off at a random starting point and tries to look at neighbouring points in the search space to improve the solution.
+It is simple and fast, it however often reaches sub-optimal solutions when there are many local optimal.
+This can be slightly avoided by starting at multiple random starting points.
+
+### Simulated Annealing
+Simulated Annealing is similar to Hill Climbing, but probabilistically accepts poorer solutions.
```

Comment:
Simulated Annealing doesn't always go for the optimal solution, but has a given probability that chooses worse solutions. This allows us to find solutions that are "deeper" in the search path.
*Added this for clarification*